File: sol/1161/1161.cpp
#include <queue>

struct Solution {
    int maxLevelSum(TreeNode* root) {
        int optSum = root->val, res = 1, lvl = 2;
        queue<TreeNode *> q;
        if (root->left != nullptr){
            q.push(root->left);
        }
        if (root->right != nullptr){
            q.push(root->right);
        }
        while (!q.empty()){
            int s = q.size(), tot = 0;
            while (s-- > 0){
                TreeNode *node = q.front();
                q.pop();
                tot += node->val;
                if (node->left != nullptr){
                    q.push(node->left);
                }
                if (node->right != nullptr){
                    q.push(node->right);
                }
            }
            if (tot > optSum){
                optSum = tot;
                res = lvl;
            }
            ++lvl;
        }
        return res;
    }
};

File: sol/1161/1161.py
from collections import deque

class Solution:
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        optSum = root.val
        res = 1
        lvl = 2
        q = deque()
        if root.left is not None:
            q.append(root.left)
        if root.right is not None:
            q.append(root.right)
        while (s := len(q)) != 0:
            tot = 0
            for _ in range(s):
                node = q.popleft()
                tot += node.val
                if node.left is not None:
                    q.append(node.left)
                if node.right is not None:
                    q.append(node.right)
            if tot > optSum:
                optSum = tot
                res = lvl
            lvl += 1
        return res
